### PR TITLE
Update deep learning guide

### PR DESCRIPTION
**Pull request by liligwu**

### CODE REVIEW

File: docs/how_to/introduction_to_deep_learning_guide.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Introduction to Deep Learning Guide
```
Comment (saadrahim) [inline docs/how_to/introduction_to_deep_learning_guide.md:1]: This type of introductory content on deep learning is not the suitable for documentation. The assumption is the end user knows what ML is. If not, there are other resources that can help them understand ML.

**Comment by Rmalavally** [comment]
Typo in Optimization - pull/2124

Have two sections for Training and Inference (rather than Optimization?)

File: docs/how_to/introduction_to_deep_learning_guide.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+:name: CommonExamplesofDeepLearning
+:widths: auto
+| Deep Learning Examples | Usage |
+|------------------------|-------|
+| Autonomous Driving | Combining deep data (maps, satellite traffic images, weather reports, a user's accumulated preferences), real-time sensor input from the environment (a deer on the road, a swerving driver), and compute power to make decisions (slow down, turn the steering wheel). |
+| Medical | Cancer research—for example, learning to detect melanoma in photos. |
+| Smart Home | Smart speakers use intelligent personal assistants and voice-recognition algorithms to comprehend and respond to unique users' verbal requests. |
+
+## Why Use AMD GPUs for Deep Learning?
+
```
Comment (saadrahim) [inline docs/how_to/introduction_to_deep_learning_guide.md:42]: This content belongs in amd.com websites outside of documentation. The proper connection is to the marketing team is  being made offline.

File: docs/examples/optimization.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
+## Optimization
```
Comment (saadrahim) [inline docs/examples/optimization.md:1]: markdown header levels are wrong here. Please pass linting.

**Review by saadrahim — CHANGES_REQUESTED** [review]
Please pass the markdown linter

File: docs/how_to/introduction_to_deep_learning_guide.md
[the remark this answers is not in the document]
Comment (liligwu) [reply docs/how_to/introduction_to_deep_learning_guide.md:1]: @amathews-amd, what's your comment please?

[the remark this answers is not in the document]
Comment (amathews-amd) [reply docs/how_to/introduction_to_deep_learning_guide.md:1]: I am okay if the main content if fully captured.

File: docs/examples/optimization.md [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+        float* results = reinterpret_cast<float*>(outputs[0].data());
+        float* max     = std::max_element(results, results + 1000);
+        int answer = max - results;
+        std::cout << "answer: " << answer << std::endl;
+    }
+    ```
+
+2. To compile this program, you can use CMake and you only need to link the `migraphx::c` library to use MIGraphX's C++ API. The following is the `CMakeLists.txt` file that can build the earlier example:
+
+    ```py
```
[the remark this answers is not in the document]
Comment (liligwu) [reply docs/examples/optimization.md]: syntax highlights corrected.

File: docs/examples/optimization.md [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+
+    message("source file: " ${EXAMPLE}.cpp " ---> bin: " ${EXAMPLE})
+    add_executable(${EXAMPLE} ${EXAMPLE}.cpp)
+
+    target_link_libraries(${EXAMPLE} migraphx::c)
+    ```
+
+3. To build the executable file, run the following from the directory containing the `inception_inference.cpp` file:
+
+    ```py
```
[the remark this answers is not in the document]
Comment (liligwu) [reply docs/examples/optimization.md]: syntax highlights corrected.

File: docs/examples/troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+:::{note}
+    Recompile PyTorch with the right gfx target if compiling from the source if the hardware is not supported. For wheels or Docker installation, contact ROCm support [6].
+:::
+
+**Q: Why am I unable to access Docker or GPU in user accounts?**
+
+Ans: Ensure that the user is added to docker, video, and render Linux groups as described in the ROCm Installation Guide at [https://docs.amd.com](https://docs.amd.com).
+
+**Q: Which consumer GPUs does ROCm support?**
```
Comment (saadrahim) [inline docs/examples/troubleshooting.md:39]: Please remove this. section

File: docs/examples/troubleshooting.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+**Q: Can I install PyTorch directly on bare metal?**
+
+Ans: Bare-metal installation of PyTorch is supported through wheels. Refer to Option 2: Install PyTorch Using Wheels Package in the section [Installing PyTorch](/ROCm/docs/how_to/pytorch_install/pytorch_install) of this guide for more information.
+
+**Q: How do I profile PyTorch workloads?**
+
+Ans: Use the PyTorch Profiler \[6\] to profile GPU kernels on ROCm.
+
+**Q: Can I run ROCm on Windows?**
```
[the remark this answers is not in the document]
Comment (liligwu) [reply docs/examples/troubleshooting.md:51]: removed

File: docs/examples/troubleshooting.md (hunk 6 above)
Comment (liligwu) [reply docs/examples/troubleshooting.md]: removed

File: docs/examples/optimization.md [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+sudo apt update && sudo apt install -y migraphx
+```
+
+The header files and libraries are installed under `/opt/rocm-\<version\>`, where \<version\> is the ROCm version.
+
+#### Option 2: Building from Source
+
+There are two ways to build the MIGraphX sources.
+
+- [Use the ROCm build tool](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX#use-the-rocm-build-tool-rbuild) - This approach uses [rbuild](https://github.com/RadeonOpenCompute/rbuild) to install the prerequisites and build the libraries with just one command.
```
[the remark this answers is not in the document]
Comment (saadrahim) [reply docs/examples/optimization.md:57]: Please ignore the last comment.

**Comment by liligwu** [comment]
@saadrahim. Would you please merge the PR if there would be no more comments? Thank you.

**Comment by saadrahim** [comment]
> @saadrahim. Would you please merge the PR if there would be no more comments? Thank you.

Hi Li,
There are still changes I would like to see. I don't need your contribution on the changes. I need to find a writer with sufficient time to add the functionality. As an example, the document name and nav organization structure of the your commit needs revision. This should be in by early next week.

**Comment by Naraenda** [comment]
@saadrahim Hi! Please re-review:
- Rebased due to conflicting changes.
- Uses footnotes for references.
  - Also removed `references.md`.
- Renamed "Examples" to "Tutorials & Examples" in TOC.
- Uses `all.md` filenames as is common with other directories.
- Converted machine learning landing page to grids.
- Minor markup improvements.

**Review by saadrahim — APPROVED** [review]

**Review by saadrahim — APPROVED** [review]

**Review by saadrahim — APPROVED** [review]

**Pull request merged**